src/power_grid_model/_utils.py

```diff
@@ -418,7 +418,7 @@ def compatibility_convert_row_columnar_dataset(
         )
 
         if is_sparse(data[comp_name]):
-            result_data[comp_name] = {"indptr": _extract_indptr(data), "data": converted_sub_data}
+            result_data[comp_name] = {"indptr": _extract_indptr(data[comp_name]), "data": converted_sub_data}
```

Comment:
I will create a separate PR where tests are added for this minor bug. I will link it once I make it. Please don't resolve this comment or merge this PR until I the bug-fix PR is merged.

Reply:
Fix in #779